test/com/google/enterprise/gsafeed/AclTest.java

```diff
@@ -41,4 +50,107 @@ public void testAcl() throws Exception {
         .build();
     assertFalse(diff.toString(), diff.hasDifferences());
   }
+
+  @Test
+  public void testGetInheritanceTypeChildOverrides() throws Exception {
+    Acl acl = unmarshal("<acl inheritance-type='child-overrides'/>");
+    assertEquals(Acl.InheritanceType.CHILD_OVERRIDES, acl.getInheritanceType());
+  }
+
+  @Test
+  public void testGetInheritanceTypeParentOverrides() throws Exception {
+    Acl acl = unmarshal("<acl inheritance-type='parent-overrides'/>");
+    assertEquals(Acl.InheritanceType.PARENT_OVERRIDES, acl.getInheritanceType());
+  }
+
+  @Test
+  public void testGetInheritanceTypeAndBothPermit() throws Exception {
+    Acl acl = unmarshal("<acl inheritance-type='and-both-permit'/>");
+    assertEquals(Acl.InheritanceType.AND_BOTH_PERMIT, acl.getInheritanceType());
+  }
+
+  @Test
+  public void testGetInheritanceTypeLeafNode() throws Exception {
+    Acl acl = unmarshal("<acl inheritance-type='leaf-node'/>");
+    assertEquals(Acl.InheritanceType.LEAF_NODE, acl.getInheritanceType());
+  }
+
+  @Test
+  public void testGetInheritanceTypeNotSet() throws Exception {
+    Acl acl = unmarshal("<acl/>");
+    assertEquals(null, acl.getInheritanceType());
+  }
+
+  @Test
+  public void testGetInheritanceTypeInvalid() throws Exception {
+    thrown.expect(IllegalArgumentException.class);
```

Comment:
If this exception isn't thrown until the getter is called, then move these down to make that more clear (and not spuriously pass the test if unmarshal throws an IllegalArgumentException). Otherwise, remove the call to the getter.